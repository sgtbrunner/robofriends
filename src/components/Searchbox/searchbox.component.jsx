import React from 'react';

const SearchBox = ({ searchChange }) => (
  <div>
    <input
      className="pa3 ba b--gray bg-washed-yellow"
      type="search"
      placeholder="find your robot here..."
      onChange={searchChange}
    />
  </div>
);

export default SearchBox;
