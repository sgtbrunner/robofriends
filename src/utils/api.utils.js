export const USERS_URL = 'https://jsonplaceholder.typicode.com/users';

export default {
  getUsers: () =>
    fetch(USERS_URL).then((response) => {
      if (response.ok) return response.json();
      throw new Error('Something went wrong');
    }),
};
