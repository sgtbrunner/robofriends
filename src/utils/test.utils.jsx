import React from 'react';
import { render } from '@testing-library/react';

export const mountComponent = ({ component, props = {} }) => {
  const Component = component;
  return render(<Component {...props} />);
};

export default null;
