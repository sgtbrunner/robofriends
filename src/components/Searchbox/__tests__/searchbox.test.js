import { screen } from '@testing-library/react';

import { mountComponent } from '../../../utils/test.utils';
import Searchbox from '../index';

describe('Searchbox component', () => {
  it('should match snapshot', () => {
    const { container } = mountComponent({
      component: Searchbox,
      props: { searchChange: jest.fn() },
    });
    expect(container).toMatchSnapshot();
  });

  it('check component placeholder text', () => {
    mountComponent({ component: Searchbox });
    const searchBox = screen.getByRole('searchbox');
    expect(searchBox).toBeInTheDocument();
  });
});
