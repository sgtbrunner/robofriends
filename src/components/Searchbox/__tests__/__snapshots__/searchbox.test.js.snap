// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbox component should match snapshot 1`] = `
<div>
  <div>
    <input
      class="pa3 ba b--gray bg-washed-yellow"
      placeholder="find your robot here..."
      type="search"
    />
  </div>
</div>
`;
